scripts/test/Muon/loadfile_presenter_test.py

```diff
@@ -0,0 +1,510 @@
+        self.mock_browse_button_to_return_files(["C:/dir1/file1.nxs", "C:/dir2/file2.nxs"])
+        self.mock_model_to_load_workspaces([[1], [2]], [1234, 1235])
+
+        self.presenter.on_browse_button_clicked()
+        self.Runner(self.presenter._load_thread)
+
+        self.assertEqual(self.view.warning_popup.call_count, 1)
+
+    @run_test_with_and_without_threading
+    def test_warning_shown_if_multiple_files_entered_by_user_in_single_file_mode(self):
+        self.mock_user_input_text("C:/dir1/file1.nxs;C:/dir2/file2.nxs")
+        self.mock_model_to_load_workspaces([[1], [2]], [1234, 1235])
+
+        self.presenter.handle_file_changed_by_user()
+        self.Runner(self.presenter._load_thread)
+
+        self.assertEqual(self.view.warning_popup.call_count, 1)
+
+    @run_test_with_and_without_threading
+    def test_single_file_from_browse_loaded_into_model_and_view_in_single_file_mode(self):
+        self.mock_browse_button_to_return_files(["C:/dir1/file1.nxs"])
+        self.mock_model_to_load_workspaces([[1]], [1234])
+        self.view.set_file_edit = mock.Mock()
+
+        self.presenter.on_browse_button_clicked()
+        self.Runner(self.presenter._load_thread)
+
+        self.assertEqual(self.model.loaded_filenames, ["C:/dir1/file1.nxs"])
+        self.assertEqual(self.model.loaded_workspaces, [[1]])
+        self.assertEqual(self.model.loaded_runs, [1234])
+
+        self.view.set_file_edit.assert_called_once_with("C:/dir1/file1.nxs", mock.ANY)
+
+    @run_test_with_and_without_threading
+    def test_single_file_from_user_input_loaded_into_model_and_view_in_single_file_mode(self):
+        self.view.set_file_edit = mock.Mock()
+        self.mock_model_to_load_workspaces([[1]], [1234])
+        self.mock_user_input_text("C:/dir1/file1.nxs")
+
+        self.presenter.handle_file_changed_by_user()
+        self.Runner(self.presenter._load_thread)
+
+        self.assertEqual(self.model.loaded_filenames, ["C:/dir1/file1.nxs"])
+        self.assertEqual(self.model.loaded_workspaces, [[1]])
+        self.assertEqual(self.model.loaded_runs, [1234])
+
+        self.view.set_file_edit.assert_called_once_with("C:/dir1/file1.nxs", mock.ANY)
+
+    @run_test_with_and_without_threading
+    def test_that_if_invalid_file_selected_in_browser_view_does_not_change(self):
+        self.mock_browse_button_to_return_files(["not_a_file"])
+        self.mock_model_to_load_workspaces([[1]], [1234])
+
+        self.view.set_file_edit = mock.Mock()
+        self.view.reset_edit_to_cached_value = mock.Mock()
+
+        self.presenter.on_browse_button_clicked()
+        self.Runner(self.presenter._load_thread)
+
+        self.model.load_workspace_from_filename = mock.Mock(side_effect=self.load_failure)
+
+        cc = self.view.set_file_edit.call_count
+        self.presenter.on_browse_button_clicked()
+        self.Runner(self.presenter._load_thread)
+
+        self.assertEqual(self.view.set_file_edit.call_count, cc)
```

Comment:
we could put this assert before the second call to the browse button

Reply:
No, because it would then not be testing the assertion that the `set_file_edit` has not been called after the second browse was triggered. I have renamed the `cc` variable to be more descriptive on this.

Reply:
ah ok